plugins/catalog-backend/migrations/20220616202842_refresh_keys.js

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+exports.up = async function up(knex) {
+  await knex.schema.createTable('refresh_keys', table => {
+    table.comment(
+      'This table contains relations between entities and keys to trigger refreshes with',
+    );
+    table
+      .text('entity_ref')
+      .notNullable()
+      .comment('A reference to the entity that the refresh key is tied to');
+    table
+      .text('key')
+      .notNullable()
+      .comment(
+        'A reference to a key which should be used to trigger a refresh on this entity',
+      );
+    table.unique(['entity_ref', 'key']);
```

Comment:
We should delete this line and replace with an index like we do here 
https://github.com/backstage/backstage/blob/7187adfadd7d06498e015f3ac26eaaf5ebc8c01d/plugins/catalog-backend/migrations/20210302150147_refresh_state.js#L168

We also need one index for the key column since we are going to find by key.

Reply:
If we removes this wouldnt it mean that the same record can get inserted everytime the processors runs? So possibly on every refresh ?
I wanted to avoid this by adding a unique constraint here and handling it on the insert by onconflict ignore.

Reply:
We'll also want to delete all existing entries for the given entity whenever we write a processing result, so that should be np